centreon/www/class/centreonGraph.class.php

```diff
@@ -433,39 +444,52 @@ public function initCurveList()
             /* Create selector for reals metrics */
             if (count($l_rmEnabled)) {
                 $l_rselector =
-                    "metric_id IN (" . implode(
+                    implode(
                         ",",
                         array_map(array("CentreonGraph", "quote"), $l_rmEnabled)
-                    ) . ")";
+                    );
                 $this->log("initCurveList with selector [real]= " . $l_rselector);
             }
             if (count($l_vmEnabled)) {
                 $l_vselector =
-                    "vmetric_id IN (" . implode(
+                    implode(
                         ",",
                         array_map(array("CentreonGraph", "vquote"), $l_vmEnabled)
-                    ) . ")";
+                    );
                 $this->log("initCurveList with selector [virtual]= " . $l_vselector);
             }
         } else {
-            /* Full Image */
             $l_rselector = "index_id = '" . $this->index . "'";
             $l_vselector = $l_rselector;
-            $this->log("initCurveList with selector= " . $l_rselector);
+            $this->log("initCurveList with selector= $l_rselector");
         }
 
         /* Manage reals metrics */
         if (isset($l_rselector)) {
-            $DBRESULT = $this->DBC->query(
-                "SELECT host_id, service_id, metric_id, metric_name, unit_name, replace(format(warn,9),',','') warn,
-                    replace(format(crit,9),',','') crit
-                    FROM metrics AS m, index_data AS i
-                    WHERE index_id = id
-                    AND " . $l_rselector . "
-                    AND m.hidden = '0'
-                    ORDER BY m.metric_name"
-            );
-            while ($rmetric = $DBRESULT->fetch()) {
+            $queryCondition = "metric_id IN (:l_rselector)";
+            if (!isset($this->metricsEnabled) || count($this->metricsEnabled) <= 0) {
+                $queryCondition = "index_id = :index_id";
+            }
+
+            $query = "SELECT host_id, service_id, metric_id, metric_name, unit_name, replace(format(warn,9),',','') warn,
+                replace(format(crit,9),',','') crit
+                FROM metrics AS m, index_data AS i
+                WHERE index_id = id
+                AND $queryCondition
+                AND m.hidden = '0'
+                ORDER BY m.metric_name";
```

Comment:
We try to use the heredoc /nowdoc for SQL query now (`<<<'SQL' ... SQL` or ``<<<SQL ... SQL`)
and use the braces when using a variable
```suggestion
            $query = <<<SQL
                SELECT 
                    host_id, 
                    service_id, 
                    metric_id, 
                    metric_name, 
                    unit_name, 
                    replace(format(warn,9),',','') warn,
                    replace(format(crit,9),',','') crit
                FROM metrics AS m, index_data AS i
                WHERE index_id = id
                    AND {$queryCondition}
                    AND m.hidden = '0'
                ORDER BY m.metric_name
                SQL;
```